feature/experimental/policy/policy_vrf_selection/otg_tests/protocol_dscp_rules_for_vrf_selection_test/protocol_dscp_rules_for_vrf_selection_test.go

```diff
@@ -269,55 +273,72 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+
+	flow := gosnappi.NewFlow().SetName(flowName)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{src.Name + "." + args.iptype}).SetRxNames([]string{dst.Name + "." + args.iptype})
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(src.MAC)
+	outerIPHeader := flow.Packet().Add().Ipv4()
+	outerIPHeader.Src().SetValue(src.IPv4)
+	outerIPHeader.Dst().SetValue(dst.IPv4)
+	outerIPHeader.Priority().Dscp().Phb().SetValue(dscp)
+	innerIPHeader := flow.Packet().Add().Ipv4()
+	innerIPHeader.Src().SetValue("198.51.100.1")
+	innerIPHeader.Dst().Increment().SetStart("203.0.113.1").SetStep("0.0.0.1").SetCount(10000)
+
+	flow.Size().SetFixed(1024)
+	flow.Rate().SetPps(100)
+	flow.Duration().FixedPackets().SetPackets(100)
+
 	return flow
 }
 
 // testTrafficFlows verifies traffic for one or more flows.
-func testTrafficFlows(t *testing.T, args *testArgs, expectPass bool, flow ...*ondatra.Flow) {
+func testTrafficFlows(t *testing.T, args *testArgs, expectPass bool, flows ...gosnappi.Flow) {
+
+	args.top.Flows().Clear()
+	for _, flow := range flows {
+		args.top.Flows().Append(flow)
+	}
+	args.ate.OTG().PushConfig(t, args.top)
+	args.ate.OTG().StartProtocols(t)
+
+	t.Logf("*** Starting traffic ...")
+	args.ate.OTG().StartTraffic(t)
+	time.Sleep(trafficDuration)
+	t.Logf("*** Stop traffic ...")
+	args.ate.OTG().StopTraffic(t)
 
-	args.ate.Traffic().Start(t, flow...)
-	time.Sleep(60 * time.Second)
-	args.ate.Traffic().Stop(t)
 	if expectPass {
 		t.Log("Expecting traffic to pass for the flows")
 	} else {
 		t.Log("Expecting traffic to fail for the flows")
 	}
 
-	// log stats
-	t.Log("All flow LossPct: ", gnmi.GetAll(t, args.ate, gnmi.OC().FlowAny().LossPct().State()))
-	t.Log("FlowAny InPkts  : ", gnmi.GetAll(t, args.ate, gnmi.OC().FlowAny().Counters().InPkts().State()))
-	t.Log("FlowAny OutPkts : ", gnmi.GetAll(t, args.ate, gnmi.OC().FlowAny().Counters().OutPkts().State()))
-
-	flowPath := gnmi.OC().FlowAny()
-	if got := gnmi.GetAll(t, args.ate, flowPath.LossPct().State()); len(got) == 0 {
-		t.Fatalf("Flow stats count not correct, wanted > 0, got 0")
-	} else {
-		for i, lossPct := range got {
-			if (expectPass == true) && (lossPct == 0) {
-				t.Logf("Traffic for %v flow is passing as expected", flow[i].Name())
-			} else if (expectPass == false) && (lossPct == 100) {
-				t.Logf("Traffic for %v flow is failing as expected", flow[i].Name())
-
-			} else {
-				t.Fatalf("Traffic is not working as expected for flow: %v.", flow[i].Name())
-			}
-
+	top := args.ate.OTG().FetchConfig(t)
+	otgutils.LogFlowMetrics(t, args.ate.OTG(), top)
+	for _, flow := range flows {
+		t.Logf("*** Verifying %v traffic on OTG ... ", flow.Name())
+		outPkts := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().OutPkts().State())
+		inPkts := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().InPkts().State())
+		lossPct := ((outPkts - inPkts) * 100) / outPkts
```

Comment:
Pardon me, I noticed one more thing.

I'm curious about the reason for switching from checking LossPct() telemetry to calculating it from OutPkts() and InPkts()? I've seen it elsewhere; maybe there's a good reason such as it not being available.

Supposing you're sticking with the formula, can you add a separate check for outPkts being zero, to have a better error message in case of division by zero?

Reply:
Yes, Valid point. I have added the check, let me know if the check is good enough.